src/NuGet.Core/NuGet.Commands/RestoreCommand/RestoreResult.cs

```diff
@@ -140,9 +145,16 @@ public void Commit(ILogger log, bool forceWrite)
                 {
                     log.LogDebug($"Writing tool lock file to disk. Path: {toolRestoreResult.LockFilePath}");
 
-                    var lockFileDirectory = Path.GetDirectoryName(toolRestoreResult.LockFilePath);
-                    Directory.CreateDirectory(lockFileDirectory);
-                    lockFileFormat.Write(toolRestoreResult.LockFilePath, toolRestoreResult.LockFile);
+                    await ConcurrencyUtilities.ExecuteWithFileLockedAsync(
+                        toolRestoreResult.LockFilePath,
+                        lockedToken =>
+                        {
+                            var lockFileDirectory = Path.GetDirectoryName(toolRestoreResult.LockFilePath);
+                            Directory.CreateDirectory(lockFileDirectory);
+                            lockFileFormat.Write(toolRestoreResult.LockFilePath, toolRestoreResult.LockFile);
+                            return Task.FromResult((object)null);
```

Comment:
I use Task.FromResult(true) normally, but obvious the result doesn't matter at all here